include/sbi/sbi_console.h

```diff
@@ -52,6 +54,22 @@ void sbi_console_set_device(const struct sbi_console_device *dev);
 struct sbi_scratch;
 
 int sbi_console_init(struct sbi_scratch *scratch);
+#else
+inline int sbi_print_empty(const char *fmt, ...) { return 0; }
+#define sbi_isprintable(ch) false
+#define sbi_getc() 0
+#define sbi_putc(ch)
+#define sbi_puts(str)
+#define sbi_gets(s, maxwidth, endchar)
+#define sbi_sprintf sbi_print_empty
+#define sbi_snprintf sbi_print_empty
+#define sbi_printf sbi_print_empty
+#define sbi_dprintf sbi_print_empty
+#define sbi_panic sbi_print_empty
```

Comment:
How about something like:

inline void sbi_panic_noprint(const char *fmt, ...) { sbi_hart_hang(); }
#define sbi_panic sbi_panic_noprint

Just defining this as empty print removes the "hang"

Reply:
Yes good point